arrow-fx/src/main/kotlin/arrow/fx/internal/Utils.kt

```diff
@@ -300,3 +300,7 @@ internal fun <A> asyncContinuation(ctx: CoroutineContext, cc: (Either<Throwable,
  */
 internal fun <A> IOForkedStart(fa: IOOf<A>, ctx: CoroutineContext): IO<A> =
   IO.Bind(IO.ContinueOn(IO.unit, ctx)) { fa.fix() }
+
+@Suppress("UNCHECKED_CAST")
+internal fun <A, B> A.unsafeCast(): B =
```

Comment:
I'm personally not a fan of this, and I think we should be consistent in the Arrow Fx codebase when doing this.

Currently it's hard coded casting everywhere, which I prefer so you can more explicitly see the types throughout the code (which is not easy to read or follow).

cc\\ @raulraja @pakoito